src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/Anchor.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2019 Michael Clarke
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+package com.github.mc1arke.sonarqube.plugin.ce.pullrequest.bitbucket;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.io.Serializable;
+
+public class Anchor implements Serializable
+{
+    @JsonProperty
+    private int line;
```

Comment:
Thank you for the feedback. Can you point out what benefit this will have compared to the current solution? I don't really see what makes @JsonCreator (Constructor) better than @JsonPropery on every field level.

Reply:
Setting it in the constructor means the fields can all be marked as final, so satisfies desinging for immutability where possible